src/firefly/js/visualize/ChartsTableViewPanel.jsx

```diff
@@ -42,35 +47,45 @@ import LOADING from 'html/images/gxt/loading.gif';
 
 const SCATTER = 'scatter';
 const HISTOGRAM = 'histogram';
-const OPTIONS_WIDTH = 350;
+const OPTIONS_WIDTH = 330;
 
 const PREF_CHART_TYPE = 'pref.chartType';
-const PREF_OPTIONS_SHOWN = 'pref.chartOptionsShown';
 
 const selectionBtnStyle = {verticalAlign: 'top', paddingLeft: 20, cursor: 'pointer'};
 
+function getChartType(props) {
+    if (props.chartId !== props.tblId) {
+         if (props.tblPlotData) { return SCATTER; }
+         if (props.tblHistogramData) { return HISTOGRAM; }
+    }
+    return (localStorage.getItem(PREF_CHART_TYPE) || SCATTER);
+}
+
 class ChartsPanel extends React.Component {
 
     constructor(props) {
         super(props);
         this.state = {
-            chartType: localStorage.getItem(PREF_CHART_TYPE) || SCATTER,
-            optionsShown: !(localStorage.getItem(PREF_OPTIONS_SHOWN)==='false'),
-            widthPx: 20,
-            heightPx: 20
+            chartType: getChartType(props),
+            optionsShown: !props.chartId,
+            immediateResize: false
         };
 
         const normal = (size) => {
             if (size) {
                 var widthPx = size.width - 10;
-                var heightPx = size.height - 30;
+                var heightPx = size.height;
                 //console.log('width: '+widthPx+', height: '+heightPx);
-                this.setState({widthPx, heightPx});
+                if (widthPx !== this.state.widthPx || heightPx != this.state.heightPx) {
+                    this.setState({widthPx, heightPx, immediateResize: false});
```

Comment:
In general, when debounced is used, you should check for isUnmounted condition before calling setState().  If not, there's a chance setState will be called after component unmounted.